src/Build/Logging/BinaryLogger/ProjectImportsCollectorExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Shared;
+
+namespace Microsoft.Build.Logging
+{
+    internal static class ProjectImportCollectorExtensions
+    {
+        private static string importingProject = ResourceUtilities.GetResourceString("ImportingProject");
+        private static int importingProjectFilePathStart = importingProject.IndexOf('"') + 1;
+        private static string importingProjectPrefix = importingProject.Substring(0, importingProjectFilePathStart);
+
+        public static void IncludeSourceFiles(this ProjectImportsCollector projectImportsCollector, BuildEventArgs e)
+        {
+            if (e is ProjectStartedEventArgs)
+            {
+                var projectStarted = (ProjectStartedEventArgs)e;
+                projectImportsCollector.AddFile(projectStarted.ProjectFile);
+            }
+            else if (e is BuildMessageEventArgs)
```

Comment:
Since you're using a message instead of a new `EventArgs`, will this parsing logic slow down the build?  You're processing every message right?  I thought you were going to add a new `EventArgs` type...